pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -294,6 +303,14 @@ async def _completions_create(
         else:
             tool_choice = 'auto'
 
+        # Some providers (e.g. MoonshotAI) don't yet accept `tool_choice="required"`.
+        profile_capabilities = OpenAIModelProfile.from_profile(self.profile)
+        if tool_choice == 'required' and not getattr(
+            profile_capabilities, 'openai_supports_tool_choice_required', True
+        ):
+            # Fall back to 'auto' so the request remains valid.
+            tool_choice = 'auto'
```

Comment:
I think we should be able to update the `elif not model_request_parameters.allow_text_output:` above to:

```python
elif not model_request_parameters.allow_text_output and OpenAIModelProfile.from_profile(self.profile).openai_supports_tool_choice_required:
```

`getattr` shouldn't be necessary

Reply:
fixed